src/wallet/rpcwallet.cpp

```diff
@@ -2769,6 +2857,10 @@ UniValue listunspent(const JSONRPCRequest& request)
             nMaximumCount = options["maximumCount"].get_int64();
     }
 
+    // Make sure the results are valid at least up to the most recent block
+    // the user could have gotten from another RPC command prior to now
+    pwallet->BlockUntilSyncedToCurrentChain();
```

Comment:
This should come after `assert(pwallet != nullptr);`

Reply:
Removed that line instead (we dont have similar checks anywhere else and the case is already covered by EnsureWalletIsAvailable).